### ISSUE TITLE
Illuminate\Auth\GenericUser::setAttribute()

### ISSUE THREAD

**Issue by bbrink68** [yuri-moens/laravel-ldap-auth#4]
Updated to latest version and now I am getting an error when trying to logout:

```
Call to undefined method Illuminate\Auth\GenericUser::setAttribute()
```

This occurs on L4OpenLdapUserProvider.php, I do have `'eloquent' => true` in the config I published per your new artisan command as the readme.md described, I am curious why the GenericUser error?

``` php
$user->setAttribute($user->getRememberTokenName(), $token);
```

**Comment by yuri-moens** [comment]
Does your user model implement Illuminate\Auth\UserInterface?

**Comment by bbrink68** [comment]
Indeed it does:

``` php
use Illuminate\Auth\UserInterface;
use Illuminate\Auth\Reminders\RemindableInterface;

class User extends Eloquent implements UserInterface, RemindableInterface {
    //...
    public function getRememberTokenName()
    {
        return 'remember_token';
    }
    //...
}
```

Above, I show that the proper method exists inside the User model as well.  Nothing changed on my end from last version to this one, which is strange now I get this error.

**Note:** I am on Laravel Version **4.2.x**, I updated after I initially got the error on **4.1.x** thinking it may resolve itself.

**Comment by bbrink68** [comment]
Interesting,  In `L4OpenLdapUserProvider.php` on line 138 I placed a `dd($user)` to see what sort of object I had:

``` php
object(Illuminate\Auth\GenericUser)[403]
  protected 'attributes' => 
    array (size=10)
      'id' => string '**54' (length=4)
      'username' => string 'mbrown' (length=6)
      'shift' => string '2nd Shift' (length=9)
      'firstname' => string 'Matthew' (length=7)
      'lastname' => string 'Brown' (length=5)
      'fullname' => string 'Matthew Brown' (length=13)
      'title' => string 'Admin' (length=5)
      'dept' => string 'Support' (length=7)
      'email' => string 'mbrown@*******.com' (length=20)
      'remember_token' => string 'a9GloBLYrUQ9xiDsap3fh8hx5A4Sz3psx6g0x7uoAmUuDM7u' (length=60)
```

`app/config/packages/ymo/l4-openldap/config.php`:

``` php
...
...
    // Use  DB To Retreive More Info?
    'use_db'    => false,
    'ldap_field'=> 'uid',
    'db_table'  => 'users',
    'db_field'  => 'username',

    // Return an Eloquent user instead of a GenericUser object
    'eloquent'              => true,
    'eloquent_user_model'   => 'User', // name of the User model
```

**Comment by bbrink68** [comment]
**You can mark this solved, or take any action you may see fit.**

The issue was that I had `'use_db'    => false'` in the config.  This was set to `false` in prior versions of your package and worked, but then again `getRememberTokenName()` wasn't being used. At the time I had placed empty versions of those functions in your L4OpenLdapUserProvider file to squelch the errors Laravel was throwing about it missing per the 'Implements UserInterface' Declaration [See This - Similar Issue](https://github.com/ccovey/ldap-auth/issues/30)

**tl:dr;**
I had to set `'use_db' => true` AND add `remember_token` column to my users table.

**Comment by yuri-moens** [comment]
Ah yes, might not have been entirely clear that use_db has to be true when using eloquent.

Anyway, glad it's solved.
